mscore/preferences.cpp

```diff
@@ -1424,6 +1442,13 @@ void PreferenceDialog::apply()
 
 #ifdef USE_PORTMIDI
       prefs.portMidiInput = portMidiInput->currentText();
+      prefs.portMidiOutput = portMidiOutput->currentText();
+      if (static_cast<PortMidiDriver*>(static_cast<Portaudio*>(seq->driver())->mididriver())->isSameCoreMidiIacBus(prefs.portMidiInput, prefs.portMidiOutput)) {
+            QMessageBox msgBox;
+            msgBox.setText("Warning: You used the same CoreMIDI IAC bus for input and output.  This will cause problematic loopback of MuseScore's output midi messages, since they will be sent back in to MuseScore as input, causing confusion.  To avoid this problem, use Mac OS's Audio MIDI Setup to create a dedicated virtual port for MuseScore's Midi output, restart MuseScore, and then select that port as output.  Then other programs may use that deidcated virtual port to receive MuseScore's midi input.");
```

Comment:
so I won't mention the words Mac or OS at all...and just take it as a given that they are on mac.  Maybe I'll say "via Finder" for the people who don't know where Audio MIDI Setup is.